StdNonModSeq.md

```diff
@@ -150,17 +150,999 @@ struct EqualsValue
+bool equal(const ExecutionSpace& ex,
+           const Kokkos::View<DataType1, Properties1...>& view1,                                                            (7)
+           Kokkos::View<DataType2, Properties2...>& view2, BinaryPredicate pred);
+
+template <class ExecutionSpace, class DataType1, class... Properties1,
+          class DataType2, class... Properties2, class BinaryPredicate>
+bool equal(const std::string& label, const ExecutionSpace& ex,
+           const Kokkos::View<DataType1, Properties1...>& view1,                                                            (8)
+           Kokkos::View<DataType2, Properties2...>& view2, BinaryPredicate pred);
+```
+
+### Description ###
+Returns true if the range [first1, last1) is equal to the range [first2, first2 + (last1 - first1)), and false otherwise for (1,2,3,4).
+Returns true if `view1` and `view2` are equal and false otherwise for (5,6,7,8)
+- (1,2,5,6): Uses `operator == ` to compare the ranges
+- (3,4,7,8): Uses binary predicate `pred` to compare the ranges.
+
+### Parameters and Requirements ###
+- `execspace`, the InputIterators `first1`, `last1`, `first2` and the views `view1`, `view2` similar to [Kokkos::Experimental::mismatch](#kokkosexperimentalmismatch)
+- `label`:
+  - (1,3): The default string is "Kokkos::equal_iterator_api_default"
+  - (5,7): The default string is "Kokkos::equal_view_api_default"
+- `pred`
+  ```cpp
+  template <class ValueType1, class ValueType2 = ValueType1>
+  struct IsEqualFunctor {
+
+      KOKKOS_INLINE_FUNCTION
+      bool operator()(const ValueType1& a, const ValueType2& b) const {
+        return (a == b);
+      }
+  };
+  ```
+
+### Return ###
+- `true` or `false` for (1,2,5,6) based on `operator == `.
+- `true` or `false` for (3,4,7,8) based on the BinaryPredicate `pred`
+
+### Eample ###
+```cpp
+namespace KE = Kokkos::Experimental;
+
+template <class ValueType1, class ValueType2 = ValueType1>
+struct IsEqualFunctor {
+
+  KOKKOS_INLINE_FUNCTION
+  bool operator()(const ValueType1& a, const ValueType2& b) const {
+    return (a == b);
+  }
+};
+
+auto exespace = Kokkos::DefaultExecutionSpace;
+using view_type = Kokkos::View<execspace, int*>;
+view_type a("a", 15);
+view_type b("b", 15);
+// fill a,b somehow
+
+// create functor
+IsEqualFunctor<int,int> p();
+
+bool isEqual = KE::equal(exespace, KE::begin(a), KE::end(a), KE::begin(b), KE::end(b) p);
+
+// assuming OpenMP is enabled, then you can also explicitly call
+bool isEqual = KE::equal(Kokkos::OpenMP(), KE::begin(a), KE::end(a), KE::begin(b), KE::end(b), p);
+
+```
+
+## Kokkos::Experimental::count_if
+
+```cpp
+template <class ExecutionSpace, class IteratorType, class Predicate>
+typename IteratorType::difference_type count_if(const ExecutionSpace& ex,
+                                                IteratorType first,
+                                                IteratorType last,                                             (1)
```

Comment:
same comment about the numbering (1) ... move closer to declaration

Reply:
Thanks. I aligned the numbers and moved them closer.